src/gpuarray_elemwise.c

```diff
@@ -31,6 +31,7 @@ struct _GpuElemwise {
 STATIC_ASSERT(GEN_CONVERT_F16 == GE_CONVERT_F16, same_flags_value_elem1);
 
 #define is_array(a) (ISCLR((a).flags, GE_SCALAR))
+#define is_rw_array(a) (ISSET((a).flags, GE_READ) && ISSET((a).flags, GE_WRITE))
```

Comment:
When we have not-inplace ops, I think the broadcasting of write-only output is allowed, isn't it ?

For example, something like:
```
matrix_a of shape (1, 4)
matrix_b of shape (5, 4)
matrix_c = elemwise2(matrix_a, '+', matrix_b, matrix_a, broadcast=True)
```